data-node/src/main/java/org/graylog/datanode/management/OpensearchRestClient.java

```diff
@@ -17,15 +17,28 @@
 package org.graylog.datanode.management;
 
 import org.graylog.datanode.process.OpensearchConfiguration;
+import org.graylog.security.certutil.CaService;
 import org.graylog.shaded.opensearch2.org.apache.http.HttpHost;
 import org.graylog.shaded.opensearch2.org.apache.http.auth.AuthScope;
 import org.graylog.shaded.opensearch2.org.apache.http.auth.UsernamePasswordCredentials;
 import org.graylog.shaded.opensearch2.org.apache.http.client.CredentialsProvider;
 import org.graylog.shaded.opensearch2.org.apache.http.impl.client.BasicCredentialsProvider;
 import org.graylog.shaded.opensearch2.org.opensearch.client.RestHighLevelClient;
+import org.graylog2.security.CustomCAX509TrustManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
 
 public class OpensearchRestClient {
-    public static RestHighLevelClient build(OpensearchConfiguration configuration) {
+    private static final Logger LOG = LoggerFactory.getLogger(OpensearchRestClient.class);
+
+    public static RestHighLevelClient build(final OpensearchConfiguration configuration, final CaService caService) {
```

Comment:
I believe that `CustomCAX509TrustManager` has to be a parameter of the method, not `CaService`.